mnemonic-memory-services/mnemonic-pmdk-pmem-service/src/main/java/org/apache/mnemonic/service/memory/internal/PMDKPMemServiceImpl.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.mnemonic.service.memory.internal;
+
+import org.apache.mnemonic.query.memory.EntityInfo;
+import org.apache.mnemonic.query.memory.ResultSet;
+import org.apache.mnemonic.service.computing.ValueInfo;
+import org.apache.mnemonic.service.memory.MemoryServiceFeature;
+import org.apache.mnemonic.service.memory.NonVolatileMemoryAllocatorService;
+import org.flowcomputing.commons.primitives.NativeLibraryLoader;
+import org.flowcomputing.commons.resgc.ReclaimContext;
+
+import java.nio.ByteBuffer;
+import java.util.HashSet;
+import java.util.Set;
+
+public class PMDKPMemServiceImpl implements NonVolatileMemoryAllocatorService {
+  private static boolean nativeLoaded = false;
+  static void loadNativeLibrary() {
+    try {
+      NativeLibraryLoader.loadFromJar("pmdkpmemallocator");
+    } catch (Exception e) {
+      throw new Error(e);
+    }
+    nativeLoaded = true;
+  }
+
+  @Override
+  public String getServiceId() {
+    return "pmdk_pmem";
+  }
+
+  @Override
+  public long init(long capacity, String uri, boolean isnew) {
+    if (!nativeLoaded) {
+      loadNativeLibrary();
+    }
+    return ninit(capacity, uri, isnew);
+  }
+
+  @Override
+  public long adjustCapacity(long id, long reserve) {
+    throw new UnsupportedOperationException("Unsupported to reduce capacity of this memory service");
+  }
+
+  @Override
+  public void close(long id) {
+    nclose(id);
+  }
+
+  @Override
+  public void syncToVolatileMemory(long id, long addr, long length, boolean autodetect) {
+    nsync(id, addr, length, autodetect);
+  }
+
+  @Override
+  public long capacity(long id) {
+    return ncapacity(id);
+  }
+
+  @Override
```

Comment:
It s better add class level documentation or inherit java docs from super class.

Reply:
sure, will add it.

Reply:
Looks Javadoc will duplicate the overriding method comment from its super class as default.